### PR TITLE
Permissions for alert suppression in machine learning rules

### PR DESCRIPTION
**Pull request by joepeeples**
Resolves https://github.com/elastic/security-docs/issues/5492.

## Previews
- ESS:
  - [Machine learning job and rule requirements](https://security-docs_bk_5819.docs-preview.app.elstc.co/guide/en/security/master/ml-requirements.html) — Added index requirements.
  - [Alert suppression](https://security-docs_bk_5819.docs-preview.app.elstc.co/guide/en/security/master/alert-suppression.html) — Updated requirements callout with link to ML requirements.

- Serverless: Click the link in the [comment below](https://github.com/elastic/security-docs/pull/5819#issuecomment-2380017654), then in the Security card click **View serverless docs**. Navigate to the pages listed above.

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-pjig2kzaf-elastic-dev.vercel.app

* Built with commit 417e4d8cb3086dcc024dcf6e0df780b13852a8b4

> Issues? Visit #next-docs in Slack

File: docs/detections/alert-suppression.asciidoc [hunk 1]
```diff
@@ -4,7 +4,9 @@
 .Requirements and notices
 [sidebar]
 --
-Alert suppression requires a https://www.elastic.co/pricing[Platinum or higher subscription].
+* Alert suppression requires a https://www.elastic.co/pricing[Platinum or higher subscription].
```
Comment (yctercero) [inline docs/detections/alert-suppression.asciidoc:7]: Platinum only for ML I believe. 

@approksiu can you confirm the licensing requirements for suppression?

Comment (approksiu) [reply docs/detections/alert-suppression.asciidoc:7]: Alert suppression is a platinum feature. So not only for ML.

**Review by rylnd — APPROVED** [review]
Looks good to me!

**Review by nastasha-solomon — APPROVED** [review]
lgtm!

**Pull request merged**